docs/guides/component-testing/component-framework-configuration.mdx

```diff
@@ -45,26 +72,100 @@ resolves to a `devServer` instance.
+trigger a request for `[devServerPublicPathRoute]/index.html` when a test is
+started. Your server needs to reply with the html-file referenced in
+`cypressConfig.indexHtmlFile` and inject a script to load the support files and
+the actual test.
 
-By default, Cypress renders your components into an HTML file located at
-`cypress/support/component-index.html`.
+```ts
+function createServer(cypressConfig, bundleDir, port = 1234) {
+  const app = express()
+
+  // read kickstart script - see below for an example
+  const clientScript = readFileSync(
+    path.join(__dirname, './client-script.js'),
+    'utf8'
+  )
+
+  app.get(
+    cypressConfig.devServerPublicPathRoute + '/index.html',
+    async (_req, res) => {
+      // read custom index.html file
+      const html = await fs.readFile(
+        path.join(cypressConfig.repoRoot, cypressConfig.indexHtmlFile),
+        { encoding: 'utf8' }
+      )
+
+      // inject kickstart-script
+      const output = html.replace(
+        '</head>',
+        `<script type="module">${clientScript}</script></head>`
+      )
+      res.send(output)
+    }
+  )
+
+  // you need to establish some url-to-path-mapping, if your bundler outputs
+  // the full directory structure you can map this one to one
+  app.use(cypressConfig.devServerPublicPathRoute, express.static(bundleDir))
+
+  app.listen(port)
+}
+```
 
-The index file allows you to add in global assets, such as styles, fonts, and
-external scripts.
+For a real-world example, you can refer to
+[this loader](https://github.com/cypress-io/cypress/blob/466155c2125476374d9f9549530f67d0c6354a41/npm/vite-dev-server/src/plugins/cypress.ts#L82-L92)
+used by the vite-devserver.
 
-You can provide an alternative path to the file using the `indexHtmlFile` option
-in the [component config](/guides/references/configuration#component) options:
+The client script must retrieve information on the currently active test from
+the Cypress instance of the parent frame and load the corresponding bundle. If a
+support file is defined, it should be injected at the top of your test bundle or
+loaded before the test script.
 
-```js
-{
-  component: {
-    devServer,
-    indexHtmlFile: '/custom/path/to/component-index.html'
-  }
+```ts
+const CypressInstance = (window.Cypress = parent.Cypress)
+const devServerPublicPathRoute = CypressInstance.config(
+  'devServerPublicPathRoute'
+)
+
+let importPromise = Promise.resolve()
+
+// If you do not bundle your support file along with the tests,
+// you need to add a separate import statement for the support file.
+const supportFilePath = CypressInstance.config('supportFile')
+if (supportFilePath) {
+  const relative = supportFilePath.replace(
+    CypressInstance.config('projectRoot'),
+    ''
+  )
+  importPromise = importPromise.then(
+    () => import(`${devServerPublicPathRoute}${relative}`)
+  )
 }
+
+// load the testfile - you can extend the load function to also load css
```

Comment:
```suggestion
// load the spec - you can extend the load function to also load css
```

Reply:
Just to clarify:

Spec - a file on your machine
Suite - a `describe`
Test - an `it`